File: server_integration_tests/java/java_server/src/main/java/dev/graphlink/javaserver/impl/Mock.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.types.Product;

import java.util.ArrayList;
import java.util.List;

/** Deterministic synthesized data for the hoist-args resolvers. */
final class Mock {
    private Mock() {}

    static List<Product> products(String prefix, int count, int offset) {
        List<Product> list = new ArrayList<>();
        for (int i = 0; i < count; i++) {
            int n = offset + i;
            list.add(Product.builder().id(prefix + "-" + n).name(prefix + "-" + n).build());
        }
        return list;
    }
}

File: server_integration_tests/kotlin/kotlin_server_suspend/src/main/kotlin/dev/graphlink/kotlinserversuspend/impl/Mock.kt
package dev.graphlink.kotlinserversuspend.impl

import dev.graphlink.kotlinserversuspend.generated.types.Product

/** Deterministic synthesized data for the hoist-args resolvers. */
internal object Mock {
    fun products(prefix: String, count: Int, offset: Int): List<Product> =
        (0 until count).map { i ->
            val n = offset + i
            Product(id = "$prefix-$n", name = "$prefix-$n")
        }
}
